### PR TITLE
TRUNK-5891: ProgramWorkflowState Domain - Switching from Hibernate Mappings to Annotations

### PR DESCRIPTION
**Pull request by alinmihaila99**
## Issue I worked on
see https://issues.openmrs.org/browse/TRUNK-5891

## Checklist: I completed these to help reviewers :)
- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by HerbertYiga** [comment]
Some  failing tests on this,checkout the CI logs

**Comment by alinmihaila99** [comment]
@HerbertYiga I saw. I am trying to resolve the problems.

**Comment by alinmihaila99** [comment]
Hi @Ruhanga, can you have a look at my commits? I really don't know why the travis fail, I tried all posibilities for the ID, but the travis still fails. All tests on my computer passed every time.

File: api/src/main/java/org/openmrs/ProgramWorkflowState.java [hunk 1]
```diff
@@ -20,14 +33,23 @@
 	// Properties
 	// ******************
 	
+	@Id
+	@Column(name = "program_workflow_state_id")
+	@GeneratedValue(strategy = GenerationType.AUTO)
```
Comment (Ruhanga) [inline api/src/main/java/org/openmrs/ProgramWorkflowState.java:38]: @alinmihaila99, can you run the `ValidateHibernateMappingsDatabaseIT` test locally. You might want to change the generation type to IDENTITY.

Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/ProgramWorkflowState.java:38]: I used the generation type IDENTITY in my second commit, but the travis failed. I also ran the ValidateHibernateMappingsDatabaseIT test locally and it passed.

**Pull request closed without merging**